src/Interfaces/Typeahead/Interface.purs

```diff
@@ -269,18 +261,71 @@ mkSingleTypeaheadMounter component inputTransformer = mkEffectFn2 \el ext -> do
         io.query $ Reset unit
     }
 
-mountSingleTypeahead :: EffectFn2 HTMLElement ExternalInput (Interface MessageVariant QueryRow)
-mountSingleTypeahead = mkSingleTypeaheadMounter single externalInputToSingleInput
-
-mountHeaderTypeahead :: EffectFn2 HTMLElement SearchDropdownInput (Interface MessageVariant QueryRow)
-mountHeaderTypeahead = mkSingleTypeaheadMounter single' searchDropdownInputToHeaderSingleInput
-
-mountToolbarTypeahead :: EffectFn2 HTMLElement SearchDropdownInput (Interface MessageVariant QueryRow)
-mountToolbarTypeahead = mkSingleTypeaheadMounter single' searchDropdownInputToToolbarSingleInput
-
 single' :: ∀ pq item. Eq item => Component pq Maybe item Aff
 single' = base
   { runSelect: const <<< Just
   , runRemove: const (const Nothing)
   , runFilter: const
   }
+
+-- DEPRECATED
+-- Use DropdownInput with dropdownInputToSingleInput
+type SearchDropdownInput =
+  { items :: Array (Object String)
+  , placeholder :: String
+  , resetLabel :: String
+  , key :: String
+  }
+
+searchDropdownInputToHeaderSingleInput
+  :: ∀ pq m
+   . SearchDropdownInput
+  -> Input pq Maybe (Object String) m
```

Comment:
One trick for "deprecating" a function is to give it a `Warn` constraint, so that the compiler throws a warning whenever it's used.

```purescript
searchDropdownInputToHeaderSingleInput
  :: ∀ pq m
   . Warn
  => SearchDropdownInput
  -> Input pq Maybe (Object String) m
```